cfdm/test/test_DimensionCoordinate.py

```diff
@@ -107,6 +108,21 @@ def test_DimensionCoordinate_climatology(self):
         self.assertTrue(x.del_climatology())
         self.assertIsNone(x.del_climatology(None))
 
+    def test_DimensiconCoordinate_array(self):
+        """Test the `DimensionCoordinate.datetime_array` method."""
+        f = cfdm.example_field(0)
+        t = f.construct("time")
+        self.assertEqual(t.array, 31)
+
+    def test_DimensiconCoordinate_datetime_array(self):
+        """Test the `DimensionCoordinate.datetime_array` method."""
+        f = cfdm.example_field(0)
+        t = f.construct("time")
+        self.assertEqual(
+            t.datetime_array,
+            np.array([cftime.DatetimeGregorian(2019, 1, 1, 0, 0, 0, 0)]),
+        )
```

Comment:
Equivalently to the previous comment, for ease of incorporating if you agree it would be good to add:

```suggestion
        )
        self.assertEqual(t.datetime_array, t.data.datetime_array)
```
```